Meta/lint-ports.py

```diff
@@ -87,24 +87,78 @@ def read_port_dirs():
 PORT_PROPERTIES = ('port', 'version', 'files', 'auth_type')
 
 
+def resolve_script_values(value: str, props: dict) -> str:
+    """Resolve all ${...} values in a string.
+
+    Args:
+        value (str): string to resolve
+        props (dict): dict of properties to resolve from
+
+    Returns:
+        str: resolved string
+    """
+    for match in re.finditer(r'\$\{([^}]+)\}', value):
+        key = match.group(1)
+        if key in props:
+            value = value.replace(match.group(0), props[key])
+    return value
+
+
+def get_script_props(dir: str, script_name: str, props: dict = {}, depth: int = 0, max_depth: int = 10) -> dict:
+    """Parse a script file and return a dict of properties.
+
+    Args:
+        dir (str): root directory of script
+        script_name (str): name of script to parse
+        props (dict): dict of properties to resolve from
+        depth (int): current depth of recursion
+        max_depth (int): maximum depth of recursion
+
+    Returns:
+        dict: dict of properties
+    """
+    if depth > max_depth:
+        print(f"Maximum recursion depth exceeded while parsing {dir}/{script_name}")
+        return props
+
+    buffer: str = ""
+    for line in open(f"{dir}/{script_name}", 'r'):
+        # Ignore comments (search in reverse to ignore # in strings)
+        if line.rfind("#") > min(line.rfind('"'), line.rfind("'"), 0):
+            line = line[0:line.rfind("#")]
+
+        line = line.rstrip()
+        buffer += line
+
+        if "=" in buffer:
+            [key, value] = buffer.split("=", 1)
+
+            if (key.startswith(" ") or key.isspace()):
+                buffer = ""
+                continue
+
+            if (value.startswith(('"', "'")) and value.endswith(value[0])):
+                if (value.count(value[0]) % 2 != 0):
```

Comment:
This check now doesn't make a lot of sense for multiline strings, since one line earlier you checked that the value both starts _and_ ends with a quote.

Reply:
Ahh I've got you now, fixed!!

Reply:
You've now removed the ability to parse multiline strings, how would you parse a multiline `files` property?

Reply:
Sorry, Should be good now 👍